src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/srm/algebras/BDD.cs

```diff
@@ -169,163 +139,184 @@ public ulong GetMax()
             obj is BDD bdd && (this == bdd || Ordinal == bdd.Ordinal && One == bdd.One && Zero == bdd.Zero);
 
         /// <summary>
-        /// Returns a topologically sorted array of all the nodes (other than True or False) in this BDD
-        /// such that, all MTBDD leaves (other than True or False) appear first in the array
-        /// and all nonterminals with smaller ordinal appear before nodes with larger ordinal.
-        /// So this BDD itself (if different from True or False) appears last.
+        /// Returns a topologically sorted array of all the nodes (other than True or False) in this BDD such that, all
+        /// MTBDD leaves (other than True or False) appear first in the array and all non-terminals with smaller ordinal
+        /// appear before nodes with larger ordinal. So this BDD itself (if different from True or False) appears last.
         /// In the case of True or False returns the empty array.
         /// </summary>
-        public BDD[] TopSort()
+        public BDD[] TopologicalSort()
         {
             if (IsFull || IsEmpty)
                 return Array.Empty<BDD>();
 
             if (IsLeaf)
                 return new BDD[] { this };
 
-            //order the nodes according to their ordinals
-            //into the nonterminals array
-            var nonterminals = new List<BDD>[Ordinal + 1];
-            var nodes = new List<BDD>();
-            var stack = new Stack<BDD>();
-            var set = new HashSet<BDD>();
+            // Produce the sorted nodes here.
+            var sorted = new List<BDD>();
+            // Non-terminals are first grouped by their ordinal into this array.
+            var nonTerminals = new List<BDD>[Ordinal + 1];
+            // Maintain sets of nodes to visit and already visited.
+            var toVisit = new Stack<BDD>();
+            var visited = new HashSet<BDD>();
 
-            stack.Push(this);
+            toVisit.Push(this);
 
-            while (stack.Count > 0)
+            while (toVisit.Count > 0)
             {
-                var node = stack.Pop();
+                var node = toVisit.Pop();
+                // True and False are not included in the result.
                 if (node.IsFull || node.IsEmpty)
                     continue;
 
                 if (node.IsLeaf)
-                    nodes.Add(node);
+                { // MTBDD terminals can be directly added to the sorted nodes.
+                    sorted.Add(node);
+                }
                 else
-                {
-                    if (nonterminals[node.Ordinal] == null)
-                        nonterminals[node.Ordinal] = new List<BDD>();
-                    nonterminals[node.Ordinal].Add(node);
-                    if (set.Add(node.Zero))
-                        stack.Push(node.Zero);
-                    if (set.Add(node.One))
-                        stack.Push(node.One);
+                { // Non-terminals are grouped by their ordinal first.
```

Comment:
We don't put comments on the same line as open braces -- I wont mark them all.

I know this is all trivia, but it catches eyeballs. It's funny how we converge on this stuff.

Reply:
Probably not worth bothering with fixing though.